env/export.go

```diff
@@ -5,8 +5,22 @@ import (
 	"strings"
 )
 
-var posixExportLineRegex = regexp.MustCompile("\\Adeclare \\-x ([a-zA-Z_]+[a-zA-Z0-9_]*)(=\")?(.+)?\\z")
-var endsWithUnescapedQuoteRegex = regexp.MustCompile("([^\\\\]\"\\z|\\A\"\\z)")
+var (
+	posixExportLineRegex = regexp.MustCompile("\\Adeclare \\-[aAfFgiIlnrtux]+ ([a-zA-Z_]+[a-zA-Z0-9_]*)(=\")?(.+)?\\z")
+	//                                                             ^
+	//                                                             |
+	//                                           all of the available options for declare
+	endsWithUnescapedQuoteRegex = regexp.MustCompile("([^\\\\]\"\\z|\\A\"\\z)")
+
+	// There are a bunch of types of bash variable that we want to ignore, becuase supporting them
+	// is either a pain, or useless, or both.
+	disallowedDeclareOpts = []string{
```

Comment:
my general policy is to avoid using regex wherever possible - they're really tricky to debug, and require a different skillset than ordinary programming to parse, IMO.

what i might do though is turn this into a map to make the process of searching for disallowed opts slightly faster and easier to read